UniversalAlgebra/Structures/AsRecordsHoms.lagda

```diff
@@ -0,0 +1,223 @@
+---
+
+\begin{code}
+
+{-# OPTIONS --without-K --exact-split --safe #-} -- cubical #-}
+
+module Structures.AsRecordsHoms where
+
+open import Axiom.Extensionality.Propositional using () renaming (Extensionality to funext)
+open import Agda.Builtin.Equality  using    ( _≡_   ;  refl     )
+open import Agda.Primitive         using    (  _⊔_  ;  lsuc     )
+                                   renaming (  Set  to Type     )
+open import Data.Product           using    (  _,_  ;  Σ
+                                            ;  _×_  ;  Σ-syntax )
+                                   renaming ( proj₁ to fst
+                                            ; proj₂ to snd      )
+open import Level                  using    ( Level ;  Lift
+                                            ; lift  ;  lower    )
+                                   renaming ( zero  to ℓ₀       )
+open import Function.Base          using    ( _∘_   ;  id       )
+open import Relation.Binary        using    ( IsEquivalence     )
+open import Relation.Binary.PropositionalEquality
+                                   using    ( cong  ; module ≡-Reasoning
+                                            ; sym   ; trans )
+
+
+
+
+open import Overture.Preliminaries    using ( ℓ₁ ; ∣_∣ ; ∥_∥ ; _⁻¹ ; _∙_ ; 𝑖𝑑 ; Π ; Π-syntax)
+open import Overture.Inverses         using ( IsInjective ; IsSurjective ; Image_∋_)
+open import Relations.Discrete        using ( ker ; kerlift )
+open import Relations.Quotients       using ( Equivalence ; Quotient
+                                            ; 0[_]Equivalence ; ker-IsEquivalence
+                                            ; kerlift-IsEquivalence ; ⟪_⟫ ; ⌞_⌟
+                                            ; ⟪_∼_⟫-elim ; _/_ )
+open import Relations.Extensionality  using ( swelldef )
+
+open import Structures.AsRecordsBasic using ( signature ; structure ; Sig∅
+                                            ; Lift-structure ; compatible )
+
+
+private variable  α β γ ρ ρ₀ ρ₁ ρ₂ : Level
+
+open structure
+open signature
+
+-- module _ {γ : Level} (𝑨 : structure {α} 𝑅 {ρ₀} 𝐹){𝑩 : structure {β} 𝑅 {ρ₁} 𝐹}(𝑪 : structure {γ} 𝑅 {ρ₂} 𝐹) where
+
+module _ {𝑅 𝐹 : signature}(𝑨 : structure {α} {ρ₀} 𝑅 𝐹)(𝑩 : structure {β} {ρ₁} 𝑅 𝐹) where
+
+ comm-rel : (symbol 𝑅) → ((carrier 𝑨) → (carrier 𝑩)) → Type (α ⊔ ρ₀ ⊔ ρ₁)
+ comm-rel 𝑟 h = ∀ a → ((rel 𝑨) 𝑟 a) → ((rel 𝑩) 𝑟) (h ∘ a)
+
+ is-hom-rel : ((carrier 𝑨) → (carrier 𝑩)) → Type (α ⊔ ρ₀ ⊔ ρ₁)
+ is-hom-rel h = ∀ R →  comm-rel R h
+
+ comm-op : (symbol 𝐹) → ((carrier 𝑨) → (carrier 𝑩)) → Type (α ⊔ β)
+ comm-op f h = ∀ a → h (((op 𝑨) f) a) ≡ ((op 𝑩) f) (h ∘ a)
+
+ is-hom-op : ((carrier 𝑨) → (carrier 𝑩)) → Type (α ⊔ β)
+ is-hom-op h = ∀ f → comm-op f h
+
+ is-hom : ((carrier 𝑨) → (carrier 𝑩)) → Type (α ⊔ ρ₀ ⊔ β ⊔ ρ₁)
+ is-hom h = is-hom-rel h × is-hom-op h
+
+ hom : Type (α ⊔ ρ₀ ⊔ β ⊔ ρ₁)
+ hom = Σ[ h ∈ ((carrier 𝑨) → (carrier 𝑩)) ] is-hom h
+
+-- EXAMPLE.  The special case when 𝑅 = ∅ (i.e., purely algebraic structures)
```

Comment:
Good idea.  Will do so in the next revision.

Reply:
Actually, this is a quite canonical example giving a purely algebraic structure as a reduct of a general structure, so I think I'll leave it.

Reply:
Some examples really are much more than just an example. That's why there are a bunch of specific categories defined in agda-categories, because they show up again in definitions of certain structures. So when there's an active reason for an example to be there beyond "being an example", it's all good.